keras/callbacks.py

```diff
@@ -824,10 +826,12 @@ def is_indexed_slices(grad):
 
     def on_epoch_end(self, epoch, logs=None):
         logs = logs or {}
-
-        if not self.validation_data and self.histogram_freq:
+        if (not self.validation_data and not
+            (self.validation_generator and
+             self.validation_steps is not None) and
```

Comment:
This pretty much forces the users to set `validation_steps` in the fit method, but sometimes this is not necessary: if `validation_generator` is a `Sequence` object, then `Model.fit` assumes `validation_steps = len(self.validation_generator)`. Maybe you could do the same thing here?